c10/macros/Macros.h

```diff
@@ -244,6 +244,14 @@ using namespace c10::xpu;
 #define C10_ALWAYS_INLINE inline
 #endif
 
+#if defined(_MSC_VER)
+#define C10_ALWAYS_INLINE_ATTRIBUTE
+#elif __has_attribute(always_inline) || defined(__GNUC__)
+#define C10_ALWAYS_INLINE_ATTRIBUTE __attribute__((__always_inline__))
+#else
+#define C10_ALWAYS_INLINE_ATTRIBUTE
+#endif
```

Comment:
Can you please move it to a separate PR? Also, why is this needed?(any perf numbers) And why exclude clang?

Reply:
> why is this needed?(any perf numbers)

not having it prevents ForcedUnroll from working properly under -Oz. See https://github.com/pytorch/executorch/pull/5247 .

> why exclude clang

clang defines __GNUC__.

Reply:
> separate PR

done. https://github.com/pytorch/pytorch/pull/136445